store/cachekv/store_bench_test.go

```diff
@@ -11,37 +9,87 @@ import (
 	"github.com/cosmos/cosmos-sdk/store/dbadapter"
 )
 
-func benchmarkCacheKVStoreIterator(numKVs int, b *testing.B) {
+// This benchmark measures the time of iterator.Next() when the parent store is blank
+func benchmarkBlankParentIteratorNext(b *testing.B, keysize int) {
+	b.StopTimer()
+
+	mem := dbadapter.Store{DB: dbm.NewMemDB()}
+	kvstore := cachekv.NewStore(mem)
+	// Use a singleton for value, to not waste time computing it
+	value := randSlice(32)
+	// Use simple values for keys, pick a random start,
+	// and take next b.N keys sequentially after.]
+	startKey := randSlice(32)
+
+	// Add 1 to avoid issues when b.N = 1
+	keys := generateSequentialKeys(startKey, b.N+1)
+	for _, k := range keys {
+		kvstore.Set(k, value)
+	}
+
 	b.ReportAllocs()
+	b.StartTimer()
+
+	iter := kvstore.Iterator(keys[0], keys[b.N])
+
+	for _ = iter.Key(); iter.Valid(); iter.Next() {
+		// TODO: Check if we need to ensure this isn't getting deadcode eliminated
+	}
+
+	iter.Close()
+}
+
+// Benchmark setting New keys to a store, where the new keys are in sequence.
+func benchmarkBlankParentAppend(b *testing.B, keysize int) {
+	b.StopTimer()
+
 	mem := dbadapter.Store{DB: dbm.NewMemDB()}
-	cstore := cachekv.NewStore(mem)
-	keys := make([]string, numKVs)
+	kvstore := cachekv.NewStore(mem)
 
-	for i := 0; i < numKVs; i++ {
-		key := make([]byte, 32)
-		value := make([]byte, 32)
+	// Use a singleton for value, to not waste time computing it
+	value := randSlice(32)
+	// Use simple values for keys, pick a random start,
+	// and take next b.N keys sequentially after.
+	startKey := randSlice(32)
 
-		_, _ = rand.Read(key)
-		_, _ = rand.Read(value)
+	keys := generateSequentialKeys(startKey, b.N)
 
-		keys[i] = string(key)
-		cstore.Set(key, value)
+	b.ReportAllocs()
+	b.StartTimer()
+
+	for _, k := range keys {
+		kvstore.Set(k, value)
 	}
+}
 
-	sort.Strings(keys)
+// Benchmark setting New keys to a store, where the new keys are random.
+// the speed of this function does not depend on the values in the parent store
+func benchmarkRandomSet(b *testing.B, keysize int) {
+	b.StopTimer()
 
-	for n := 0; n < b.N; n++ {
-		iter := cstore.Iterator([]byte(keys[0]), []byte(keys[numKVs-1]))
+	mem := dbadapter.Store{DB: dbm.NewMemDB()}
+	kvstore := cachekv.NewStore(mem)
+
+	// Use a singleton for value, to not waste time computing it
+	value := randSlice(32)
+	keys := generateRandomKeys(keysize, b.N)
 
-		for _ = iter.Key(); iter.Valid(); iter.Next() {
-		}
+	b.ReportAllocs()
+	b.StartTimer()
```

Comment:
We can skip using b.StopTimer() and then b.StartTimer() in favour of just b.ResetTimer() here.